core/node/groups.go

```diff
@@ -281,9 +281,10 @@ func Online(bcfg *BuildCfg, cfg *config.Config) fx.Option {
 
 	/* don't provide from bitswap when the strategic provider service is active */
 	shouldBitswapProvide := !cfg.Experimental.StrategicProviding
+	pbrf := bcfg.PeerBlockRequestFilter
 
 	return fx.Options(
-		fx.Provide(OnlineExchange(cfg, shouldBitswapProvide)),
+		fx.Provide(OnlineExchange(cfg, shouldBitswapProvide, pbrf)),
```

Comment:
I belive you inspired your pbrf variable from the line above, this code (`shouldBitswapProvide`) is a clarification variable, it translate `!cfg.Experimental.StrategicProviding` into a usefull thing we understand.
In your case you are turning a perfectly good `PeerBlockRequestFilter` into a meaningless without prior knowledge `pbrf` one.
```suggestion
	return fx.Options(
		fx.Provide(OnlineExchange(cfg, shouldBitswapProvide, bcfg.PeerBlockRequestFilter)),
```